lib/puppet/network/http/api.rb

```diff
@@ -1,9 +1,11 @@
 class Puppet::Network::HTTP::API
+
   def self.not_found
     Puppet::Network::HTTP::Route.
       path(/.*/).
       any(lambda do |req, res|
-        raise Puppet::Network::HTTP::Error::HTTPNotFoundError.new("No route for #{req.method} #{req.path}", Puppet::Network::HTTP::Issues::HANDLER_NOT_FOUND)
+        raise Puppet::Network::HTTP::Error::HTTPNotFoundError.new("No route for #{req.method} #{req.path} #{Puppet::Network::HTTP::NOT_FOUND_ERROR_MESSAGE}",
+                                                                  Puppet::Network::HTTP::Issues::HANDLER_NOT_FOUND)
```

Comment:
So with this change, anything that uses this helper will have this error message. So, for example, a request to `/puppet/v2.0/unknown` would get this message, even though that does have the prefix and version string, so is probably not coming from a Puppet 3 agent. Given that we're removing the v2.0 endpoints, I'm not actually sure for right now that there will be anywhere that want to be using a `not_found` helper that wouldn't want this custom error message, but I feel like it might be good to have a general not_found helper and then a more specific one with this error message?